Description: Use system Korean font instead of Nanum
 This patch removes CSS references to Nanum. As refrenced by
 LP: #2031685, we already have Korean glyph coverage in the default
 system fonts, so we want to avoid pulling fonts-nanum into main.
Author: Zixing Liu <zixing.liu@canonical.com>
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/rustc/+bug/2031685
Forwarded: not-needed
Last-Update: 2026-01-07
---
--- a/tests/rustdoc-gui/enum-variants.goml
+++ b/tests/rustdoc-gui/enum-variants.goml
@@ -6,5 +6,5 @@
 
 assert-css: (
     "details.non-exhaustive > summary",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
--- a/tests/rustdoc-gui/module-items-font.goml
+++ b/tests/rustdoc-gui/module-items-font.goml
@@ -2,75 +2,75 @@
 go-to: "file://" + |DOC_PATH| + "/test_docs/index.html"
 assert-css: (
     ".item-table dt > a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
     ALL,
 )
 assert-css: (
     ".item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
     ALL,
 )
 
 // modules
 assert-css: (
     "#modules + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#modules + .item-table ",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
 // structs
 assert-css: (
     "#structs + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#structs + .item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
 // enums
 assert-css: (
     "#enums + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#enums + .item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
 // traits
 assert-css: (
     "#traits + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#traits + .item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
 // functions
 assert-css: (
     "#functions + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#functions + .item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
 // keywords
 assert-css: (
     "#keywords + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#keywords + .item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
 // attributes
 assert-css: (
     "#attributes + .item-table dt a",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 assert-css: (
     "#attributes + .item-table dd",
-    {"font-family": '"Source Serif 4", NanumBarunGothic, serif'},
+    {"font-family": '"Source Serif 4", serif'},
 )
--- a/tests/rustdoc-gui/scrape-examples-fonts.goml
+++ b/tests/rustdoc-gui/scrape-examples-fonts.goml
@@ -1,7 +1,7 @@
 // This test ensures that the correct font is used in scraped examples.
 go-to: "file://" + |DOC_PATH| + "/scrape_examples/fn.test_many.html"
 
-store-value: (font, '"Fira Sans", Arial, NanumBarunGothic, sans-serif')
+store-value: (font, '"Fira Sans", Arial, sans-serif')
 
 wait-for-css: (".scraped-example-title", {"font-family": |font|})
 wait-for-css: (".more-examples-toggle summary", {"font-family": |font|})
--- a/tests/rustdoc-gui/toggle-docs.goml
+++ b/tests/rustdoc-gui/toggle-docs.goml
@@ -10,7 +10,7 @@
 assert-text: ("#toggle-all-docs", "Show all")
 assert-css: (
     "#main-content > details.top-doc > summary",
-    {"font-family": '"Fira Sans", Arial, NanumBarunGothic, sans-serif'},
+    {"font-family": '"Fira Sans", Arial, sans-serif'},
 )
 click: "#toggle-all-docs"
 // Not collapsed anymore so the "open" attribute should be back.
--- a/src/librustdoc/html/static/css/rustdoc.css
+++ b/src/librustdoc/html/static/css/rustdoc.css
@@ -42,7 +42,7 @@
 	--code-block-border-radius: 6px;
 	--impl-items-indent: 0.3em;
 	--docblock-indent: 24px;
-	--font-family: "Source Serif 4", NanumBarunGothic, serif;
+	--font-family: "Source Serif 4", serif;
 	--font-family-code: "Source Code Pro", monospace;
 	--line-number-padding: 4px;
 	--line-number-right-margin: 20px;
@@ -329,7 +329,7 @@
 .search-switcher,
 /* This selector is for the items listed in the "all items" page. */
 ul.all-items {
-	font-family: "Fira Sans", Arial, NanumBarunGothic, sans-serif;
+	font-family: "Fira Sans", Arial, sans-serif;
 }
 
 #toggle-all-docs,
@@ -2530,7 +2530,7 @@
 	bottom: 0;
 	z-index: 1;
 	background-color: var(--main-background-color);
-	font: 1rem/1.5 "Source Serif 4", NanumBarunGothic, serif;
+	font: 1rem/1.5 "Source Serif 4", serif;
 	/* To make it look a bit better and not have it stuck to the preceding element. */
 	padding-left: 0.2em;
 }
--- a/tests/rustdoc-gui/font-serif-change.goml
+++ b/tests/rustdoc-gui/font-serif-change.goml
@@ -2,7 +2,7 @@
 go-to: "file://" + |DOC_PATH| + "/test_docs/index.html"
 
 // By default, it should be the serif fonts.
-store-value: (serif_font, '"Source Serif 4", NanumBarunGothic, serif')
+store-value: (serif_font, '"Source Serif 4", serif')
 store-value: (serif_code_font, '"Source Code Pro", monospace')
 assert-css: ("body", {"font-family": |serif_font|})
 assert-css: ("p code", {"font-family": |serif_code_font|})
